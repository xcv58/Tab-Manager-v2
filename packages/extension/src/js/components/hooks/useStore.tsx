import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}

export const useFontSize = () => {
  const { userStore } = useStore()
  return userStore.fontSize
}

export const useTabHeight = () => {
  const fontSize = useFontSize()
  return 3 * fontSize
}
