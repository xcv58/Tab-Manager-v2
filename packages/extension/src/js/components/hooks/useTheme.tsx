import { createContext, useContext } from 'react'

export const ThemeContext = createContext(false)

export const useTheme = () => useContext(ThemeContext)
