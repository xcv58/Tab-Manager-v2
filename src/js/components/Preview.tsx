import React from 'react'
import { inject, observer } from 'mobx-react'
import Tab from 'components/Tab/Tab'
import { focusedColor } from 'libs/colors'

const style = {
  opacity: 0.5,
  background: focusedColor
}

@inject('tabStore')
@observer
class Preview extends React.Component<any> {
  render () {
    const {
      tabStore: { sources }
    } = this.props
    const content = sources.map(tab => <Tab key={tab.id} tab={tab} faked />)
    return <div style={{ ...style, ...this.props.style }}>{content}</div>
  }
}

export default Preview
