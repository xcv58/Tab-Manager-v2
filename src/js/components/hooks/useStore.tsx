import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}
