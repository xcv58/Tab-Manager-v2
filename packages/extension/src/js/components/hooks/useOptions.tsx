import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return null
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  const { historyTabs } = searchStore
  console.log({ historyTabs: [...historyTabs] })
  return [...windowStore.tabs, ...historyTabs]
}
