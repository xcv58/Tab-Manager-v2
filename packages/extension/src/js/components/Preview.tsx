import React, { CSSProperties } from 'react'
import classNames from 'classnames'
import { observer } from 'mobx-react-lite'
import { useStore } from './hooks/useStore'
import { useTheme } from './hooks/useTheme'
import ViewOnlyTab from './Tab/ViewOnlyTab'

export default observer((props: { style: CSSProperties }) => {
  const isDarkTheme = useTheme()
  const { tabStore } = useStore()
  const { sources } = tabStore
  const content = sources.map((tab) => <ViewOnlyTab key={tab.id} tab={tab} />)
  return (
    <div
      className={classNames('opacity-50', {
        'bg-blue-300': !isDarkTheme,
        'bg-gray-900': isDarkTheme,
      })}
      style={props.style}
    >
      {content}
    </div>
  )
})
