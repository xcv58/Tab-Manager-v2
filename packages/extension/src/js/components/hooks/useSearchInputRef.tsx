import { useRef, useEffect } from 'react'
import { useStore } from 'components/hooks/useStore'

export const useSearchInputRef = () => {
  const searchInputRef = useRef<HTMLInputElement>(null)
  const { searchStore } = useStore()
  const { setSearchEl } = searchStore

  useEffect(() => setSearchEl(searchInputRef))
  return searchInputRef
}
