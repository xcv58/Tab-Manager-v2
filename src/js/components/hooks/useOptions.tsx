import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  return windowStore.tabs
}
