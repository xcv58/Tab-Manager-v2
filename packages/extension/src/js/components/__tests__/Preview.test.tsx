import React from 'react'
import { render } from '@testing-library/react'
import Preview from 'components/Preview'
import * as StoreContext from 'components/hooks/useStore'

// TODO: Add real tab mock
const sources = []
const mockStore = {
  tabStore: { sources },
}

describe('Preview', () => {
  beforeEach(() => {
    jest.spyOn(StoreContext, 'useStore').mockImplementation(() => mockStore)
  })

  it('render correct components', () => {
    const { container } = render(<Preview />)
    expect(container).toMatchSnapshot()
    expect(container.querySelector('div')).toBeInTheDocument()
  })
})
