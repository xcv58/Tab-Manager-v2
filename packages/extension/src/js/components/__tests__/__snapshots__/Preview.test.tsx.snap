// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview render correct components 1`] = `
<div>
  <div
    class="opacity-50 bg-blue-300"
  />
</div>
`;
